src/unity/python/turicreate/test/test_sframe_sequence_iterator.py

```diff
@@ -59,3 +82,131 @@ def test_ceil_dev(self):
+
+    def test_next_batch_size_32_with_target(self):
+        # The chunked dataset is smaller than batch size, first batch will be padded
+        self._test_next(32 , 1 , self.expected_batches_32)
+
+    def test_next_batch_size_3(self):
+        # The chunked dataset length > batch size and < 2 * batch_size, the second batch will be padded
+        self._test_next(3 , 2 , self.expected_batches_3)
+
+    def test_next_batch_size_3_with_padding(self):
+        self._test_next(3 , 2 , self.expected_batches_3, use_pad=True)
+
+    def test_next_batch_size_4_with_padding(self):
+        # The chunked dataset length == batch size
+        self._test_next(4 , 1 , self.expected_batches_4)
+
+    def _test_next(self , batch_size , expected_num_batches , expected_batches, use_taget=True, use_pad=False):
+        chunked , num_sessions = sframe_sequence_iterator.prep_data(
+            self.data, self.features, self.session_id, self.prediction_window,
+            self.predictions_in_chunk, target=self.target)
+
+        seq_iter = sframe_sequence_iterator.SFrameSequenceIter(chunked , len(self.features) , self.prediction_window,
+                                                               self.predictions_in_chunk , batch_size ,
+                                                               use_target=use_taget, use_pad=use_pad)
+
+        self.assertTrue(seq_iter.batch_size == batch_size)
+        self.assertTrue(seq_iter.num_batches == expected_num_batches)
+
+        for batch_num ,expected_batch in enumerate(expected_batches):
+            observed_batch = seq_iter.next()
+
+            np.testing.assert_array_equal(expected_batch['features'] , observed_batch.data[0].asnumpy(),
+                                          "Error - features in batch %d" % batch_num)
+
+            if (use_taget):
+                np.testing.assert_array_equal(expected_batch['target'] , observed_batch.label[0].asnumpy(),
+                                              "Error - target in batch %d" % batch_num)
+
+                np.testing.assert_array_equal(expected_batch['weights'] , observed_batch.label[1].asnumpy(),
+                                              "Error - weights in batch %d" % batch_num)
+            else:
+                self.assertTrue(observed_batch.label is None)
+
+            if (use_pad and "padding" in expected_batch):
+                self.assertTrue(observed_batch.pad == expected_batch["padding"])
+            else:
+                self.assertTrue(observed_batch.pad == 0)
+
+        # The iterator length should be in the same length as the expected_batches list.
+        # We expect a "StopIteration" exception to be raised on the next call to next()
+        stop_iteration_raised = False
+        try:
+            seq_iter.next()
+        except StopIteration:
+            stop_iteration_raised = True
+        self.assertTrue(stop_iteration_raised , "Error - the iterator is longer than expected")
+
+    @classmethod
```

Comment:
`classmethod` should receive `cls` instead of `self`. 
On the other hand, `_prepare_expected_chunked_dataset`, `_prepare_expected_batches`, and `setUp` should not be `classmethod`.  The setup method should be `setUp` which is called before every test, or `setUpClass` which is a `classmethod` invoked once for the entire test class.

Reply:
Fixed